command.go

```diff
@@ -230,6 +231,33 @@ func (c *Command) HelpFunc() func(*Command, []string) {
 	}
 }
 
+// Output the usage for the command
+// Used when a user provides invalid input
+// Can be defined by user by overriding UsageFunc
+func (c *Command) Usage() error {
+	c.mergePersistentFlags()
+	err := tmpl(c.getOutOrStderr(), c.UsageTemplate(), c)
```

Comment:
@dnephin  Unless I'm missing something, it actually works the other way - `UsageFunc()` [uses this one](https://github.com/fabianofranz/cobra/blob/77530f537eb200c0781aac768e568bf35191525c/command.go#L208), and if another func were provided through `SetUsageFunc`, [it's going to be used](https://github.com/fabianofranz/cobra/blob/77530f537eb200c0781aac768e568bf35191525c/command.go#L200-L202) and this one will be ignored. 

I mostly did this to improve code clarity and conciseness: `UsageFunc()` calls `Usage()` the same way `HelpFunc()` were already calling `Help()`.